src/routing.rs

```diff
@@ -1,30 +1,58 @@
-use axum::extract::Path;
-use axum::Router;
-use axum::routing::get;
+use std::sync::{Arc, RwLock};
+use axum::{
+    extract::{Path, State},
+    http::StatusCode,
+    routing::{get, post},
+    response::Response,
+    Router
+};
 use rust_intro::rust_intro::DataMessage;
+use crate::data::DataHandler;
 
-pub(crate) fn init_routes() -> Router {
+pub(crate) async fn init_routes() -> Router {
+    let handler = Arc::new(RwLock::new(DataHandler::new()));
```

Comment:
Consider Wrapping the state in a struct to make it easier to add more fields to it without modifying the signature of every route.

e.g. add
```rust
#[derive(Clone)]
struct IntroState {
    handler: Arc<RwLock<DataHandler>>
}
...
let state = IntroState {
    handler: Arc::new(RwLock::new(DataHandler::new()))
};
```